src/core/exec-invoke.c

```diff
@@ -4906,7 +4908,13 @@ int exec_invoke(
                         u = NULL;
 
                 if (u) {
-                        r = get_fixed_user(u, &username, &uid, &gid, &pwent_home, &shell);
+                        /* We can't use nss unconditionally for root without risking deadlocks if some IPC services
+                         * will be started by pid1 and are ordered after us. But if SetLoginEnvironment= is
```

Comment:
Hm, I noticed that SetLoginEnvironment= is enabled by default when User= or DynamicUser= is enabled.
So, many processes may be affected with the change, like resolved, networkd, timesyncd, and so on. Is it really safe?
If an NSS requires networking,,, what happens?

Reply:
It's effectively enabled in the sense of exec_context_get_set_login_environment. The field itself is tristate.

Reply:
Ah, "enabled _explicitly_".
Could you extend the comment to mention that? Otherwise, we might mistakenly replace the condition with `exec_context_get_set_login_environment()` in a future.